lte/gateway/c/oai/tasks/nas/emm/sap/emm_as.c

```diff
@@ -925,8 +925,8 @@ static int _emm_as_establish_req(emm_as_establish_t *msg, int *emm_cause)
           "UE.ue_id=" MME_UE_S1AP_ID_FMT " \n",
           msg->ue_id);
         //Clean up S1AP and MME UE Context
-        nas_itti_detach_req(msg->ue_id);
         unlock_ue_contexts(ue_mm_context);
```

Comment:
Unlock UE context after the mme_app_handle_detach_req call is done.

Reply:
In mme_app_handle_detach_req(), if UE doesn't have active PDN session at SPGW, then MME sends Purge UE Req to HSS and UE Context Release Req to eNB and deletes the UE context. So after deleting UE context and if we try to unlock UE, it was resulting to crash.

Reply:
If UE has valid PDN session at SPGW then MME sends Delete Session Req to SPGW to delete active PDN contexts at SPGW. After the reception of Delete Session Response MME sends the Purge UE Req and UE Context Release Req. After the reception of UE context Release Complete, UE contexts are deleted. If we don't unlock it, it was failing to get lock while processing UE context release complete. So it is necessary to have lock before the  mme_app_handle_detach_req